R/dtc_create_iso8601.R

```diff
@@ -0,0 +1,415 @@
+#' sdtm.oak:::iso8601_truncate(
+#'   c(
+#'     "1999----T15:20:01",
+#'     "1999-01-01T-:20:01",
+#'     "1999-01-01T-:-:01",
+#'     "1999-01-01T-:-:-"
+#'   )
+#' )
+#'
+#' @keywords internal
+iso8601_truncate <- function(x, empty_as_na = TRUE) {
+  x <- stringr::str_remove(x, "[^\\d]*$")
+  if (empty_as_na) x[x == ""] <- NA_character_
+  x
+}
+
+#' Convert date/time components into ISO8601 format
+#'
+#' [format_iso8601()] takes a character matrix of date/time components and
+#' converts each component to ISO8601 format. In practice this entails
+#' converting years to a four digit number, and month, day, hours, minutes and
+#' seconds to two-digit numbers. Not available (`NA`) components are converted
+#' to `"-"`.
+#'
+#' @param m A character matrix of date/time components. It must have six
+#'   named columns: `year`, `mon`, `mday`, `hour`, `min` and `sec`.
+#' @param .cutoff_2000 An integer value. Two-digit years smaller or equal to
+#'   `.cutoff_2000` are parsed as though starting with `20`, otherwise parsed as
+#'   though starting with `19`.
+#'
+#' @returns A character vector with date-times following the ISO8601 format.
+#'
+#' @examples
+#' cols <- c("year", "mon", "mday", "hour", "min", "sec")
+#' m <- matrix(
+#'   c(
+#'     "99", "00", "01",
+#'     "Jan", "feb", "03",
+#'     "1", "01", "31",
+#'     "00", "12", "23",
+#'     "00", "59", "10",
+#'     "42", "5.15", NA
+#'   ),
+#'   ncol = 6,
+#'   dimnames = list(c(), cols)
+#' )
+#'
+#' sdtm.oak:::format_iso8601(m)
+#'
+#' @keywords internal
+format_iso8601 <- function(m, .cutoff_2000 = 68L) {
+  admiraldev::assert_integer_scalar(.cutoff_2000)
+
+  m[, "year"] <- iso8601_year(m[, "year"], cutoff_2000 = .cutoff_2000)
+  m[, "mon"] <- iso8601_mon(m[, "mon"])
+  m[, "mday"] <- iso8601_mday(m[, "mday"])
+  m[, "hour"] <- iso8601_hour(m[, "hour"])
+  m[, "min"] <- iso8601_min(m[, "min"])
+  m[, "sec"] <- iso8601_sec(m[, "sec"])
+
+  m <- iso8601_na(m)
+
+  x <-
+    paste0(
+      m[, "year"],
+      "-",
+      m[, "mon"],
+      "-",
+      m[, "mday"],
+      "T",
+      m[, "hour"],
+      ":",
+      m[, "min"],
+      ":",
+      m[, "sec"]
+    )
+
+  iso8601_truncate(x)
+}
+
+#' Convert date or time collected values to ISO 8601
+#'
+#' [create_iso8601()] converts vectors of dates, times or date-times to [ISO
+#' 8601](https://en.wikipedia.org/wiki/ISO_8601) format. Learn more in
+#' `vignette("iso_8601")`.
+#'
+#' @param ... Character vectors of dates, times or date-times' components.
+#' @param .format Parsing format(s). Either a character vector or a list of
+#'   character vectors. If a character vector is passed then each element is
+#'   taken as parsing format for each vector passed in `...`. If a list is
+#'   provided, then each element must be a character vector of formats. The
+#'   first vector of formats is used for parsing the first vector passed in
+#'   `...`, and so on.
+#' @param .fmt_c A list of regexps to use when parsing `.format`. Use [fmt_cmp()]
```

Comment:
Do we need to add an example on how to use fmt_cmp()

Reply:
Done: https://github.com/pharmaverse/sdtm.oak/pull/21/commits/aa2113556c29395161fddae73eea72e3dbe5ba9b.